task/src/global_registry.rs

```diff
@@ -32,12 +33,12 @@ pub struct GlobalRegistry {
     /// up-to-date shared list of statuses
     /// only used if `state_cpy` is out of date
     /// or if appending
-    state_list: Arc<RwLock<Vec<(TaskState, String)>>>,
+    state_list: Arc<RwLock<BTreeMap<HotShotTaskId, (TaskState, String)>>>,
```

Comment:
Did we rename `state_cpy` to `state_cache`? Looks like we need to update the documentation here.